### PR TITLE
[Issue h2oai/19052] Upgrade Spring Boot version to 2.3.0

### PR DESCRIPTION
**Pull request by this**
A fix for https://github.com/h2oai/h2oai/issues/19052

### CODE REVIEW

**Comment by this** [comment]
@achraf-mer We should upgrade Spring version to `2.3.0.RELEASE` (see https://docs.google.com/spreadsheets/d/1A6L8MQbx6-Ihi8F4z-5Fl6AtssrYewA9ONXKuFxp5bQ/edit#gid=703030260&range=A41:L41), but `spring-boot-gradle-plugin:2.3.0.RELEASE` requires Gradle `v5.6.x`. Refer below error message produced when building `./gradlew distributionZip`.

```
* What went wrong:
An exception occurred applying plugin request [id: 'org.springframework.boot']
> Failed to apply plugin [id 'org.springframework.boot']
   > Spring Boot plugin requires Gradle 5 (5.6.x only) or Gradle 6 (6.3 or later). The current version is Gradle 5.2.1

```

Shall we upgrade the Gradle version to `vv5.6.4`? Current Gradle version is `v5.2.1` https://github.com/h2oai/dai-deployment-templates/blob/v1.0.25/gradle/wrapper/gradle-wrapper.properties#L3

**Comment by this** [comment]
@mmalohlava How should we proceed to do a release out of this PR?

File: common/rest-jdbc-spring-api/build.gradle [hunk 1]
```diff
@@ -7,6 +7,7 @@ apply from: project(":").file('gradle/java.gradle')
 dependencies {
     implementation group: 'io.swagger', name: 'swagger-annotations'
     implementation group: 'org.springframework.boot', name: 'spring-boot-starter-web'
+    implementation group: 'org.springframework.boot', name: 'spring-boot-starter-validation'
```
Comment (nkpng2k) [inline common/rest-jdbc-spring-api/build.gradle:10]: qq - what is this for please?

**Review by nkpng2k — APPROVED** [review]
just need to know what the extra dependency is for please

File: common/rest-jdbc-spring-api/build.gradle (hunk 1 above)
Comment (this) [reply common/rest-jdbc-spring-api/build.gradle:10]: Spring Boot 2.3.0 web starter doesn't include the `javax.validation.*` package. Hence we need to include the `spring-boot-starter-validation` as a dependency https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.3-Release-Notes#validation-starter-no-longer-included-in-web-starters

Comment (this) [reply common/rest-jdbc-spring-api/build.gradle:10]: See commit b590e732bdb4544f56ba4b63632a28ef792cf94c for more info

**Review by achraf-mer — APPROVED** [review]

**Comment by mmalohlava** [comment]
@this pls go ahead with merge. 

you can release directly the package as well: follow instructions here https://github.com/h2oai/dai-deployment-templates/#step-by-step or ping me on Slack.

**Comment by mmalohlava** [comment]
thank you for the change.

**Comment by mmalohlava** [comment]
@this quick update: I spoke with @arno - i still need to release MOJO first. then will release this one and upgrade refs in DAI.

**Comment by this** [comment]
> @this quick update: I spoke with @arno - i still need to release MOJO first. then will release this one and upgrade refs in DAI.

Thanks @mmalohlava

**Pull request merged**